src/main/java/uk/gov/gchq/palisade/util/AbstractResourceBuilder.java

```diff
@@ -79,15 +81,45 @@ public static Resource create(final String uriString) {
      * @param uri the uri of the resource you want built
      * @return a newly created resource with the id of the uri.
      */
+    @SuppressWarnings("java:S1075") //Suppress hardcoded path separator code smell
     public Resource buildNormal(final URI uri) {
-        try {
-            URI normal = UriBuilder.create(uri)
+        URI absoluteResourceId;
+
+        if (!uri.getSchemeSpecificPart().startsWith("/")) {
+            var localResource = new File(uri.getSchemeSpecificPart());
+            String path;
+            try {
+                path = localResource.getCanonicalPath();
+            } catch (IOException e) {
+                LOGGER.warn("Unable to get the Canonical path value", e);
+                path = localResource.getAbsolutePath();
+            }
+
+            if (!path.startsWith("/")) {
+                path = "/" + path;
+            }
+
+            // Check if the resource is a directory and the path does not end with a "/"
+            if (localResource.isDirectory() && !path.endsWith("/")) {
+                path += "/";
+            }
+            absoluteResourceId = UriBuilder.create(uri)
+                    .withoutScheme()
+                    .withoutAuthority()
+                    .withPath(path)
+                    .withoutQuery()
+                    .withoutFragment();
+        } else {
```

Comment:
Can you parameterise out the `URI_PATH_SEPARATOR = "/"` for clarity as to why we're not concerned about forward vs backslashes?

Reply:
updated